crates/nargo_cli/tests/prove_and_verify.rs

```diff
@@ -51,13 +48,17 @@ mod tests {
 
     #[test]
     fn noir_integration() {
-        let current_dir = std::env::current_dir().unwrap();
-
-        let test_data_dir = current_dir.join(TEST_DIR).join(TEST_DATA_DIR);
+        // Try to find the directory that Cargo sets when it is running; otherwise fallback to assuming the CWD
+        // is the root of the repository and append the crate path
+        let manifest_dir = match std::env::var("CARGO_MANIFEST_DIR") {
```

Comment:
This seems to revert some of the changes by @TomAFrench in #1180 -- Both refactors make the code more readable, so I'm okay with keeping either

Reply:
This is fine and addresses https://github.com/noir-lang/noir/pull/1114/files/6a79a0e94260a78d5ccd5c25b24c657b4ebac7f2#r1162150191

This doesn't really affect my refactoring as that focused on the for-loop below.